src/Microsoft.Health.Fhir.Core/Exceptions/ResourceNotFoundException.cs

```diff
@@ -10,15 +10,18 @@ namespace Microsoft.Health.Fhir.Core.Exceptions
 {
     public class ResourceNotFoundException : FhirException
```

Comment:
Instead of extending this exception, wouldn't it be better to create a new exception called GroupNotFoundException? 

I'm suggesting that because ResourceNotFoundException is probably used in many other scenarios, and adding a property/parameter called groupNotFound makes it too specific to a single case.

Reply:
GroupNotFoundException would inherit from ResourceNotFoundException.